test/system/src/test/java/io/pravega/test/system/MultiReaderWriterWithFailOverTest.java

```diff
@@ -0,0 +1,387 @@
+        log.info("Scope passed to readergroup manager {}", scope);
+        readerGroupManager = ReaderGroupManager.withScope(scope, controllerUri);
+        readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(STREAM_NAME));
+        log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+        log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+        log.info("Online readers {}", readerGroupManager.getReaderGroup(readerGroupName).getOnlineReaders());
+
+        //create readers
+        log.info("Creating {} readers", NUM_READERS);
+        List<EventStreamReader<Long>> readerList = new ArrayList<>();
+        String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+        log.info("Scope that is seen by readers {}", scope);
+        //start reading events
+        for (int i = 0; i < NUM_READERS; i++) {
+            log.info("Starting reader{}", i);
+            log.info("Creating reader with id {}", readerName + i);
+            final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                    readerGroupName,
+                    new JavaSerializer<Long>(),
+                    ReaderConfig.builder().build());
+            readerList.add(reader);
+        }
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+        readWriteTest(writerList, readerList);
+        log.debug("{} with 2 controller, 2 SSS instances succeeds", "multiReaderWriterTest");
+
+        //Scale down SSS instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        Thread.sleep(60000);
+        log.info("Scaling down SSS instances from 3 to 2");
+
+        eventsReadFromPravega = new ConcurrentLinkedQueue<>();
+        stopReadFlag.set(false);
+        eventData = new AtomicLong(); //data used by each of the writers.
+        eventReadCount = new AtomicLong(); // used by readers to maintain a count of events.
+
+        log.info("Test with SSS failover");
+        readWriteTest(writerList, readerList);
+        log.debug("{} with SSS failover succeeds", "multiReaderWriterTest");
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        log.info("Scaling down controller instances from 3 to 2");
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                .collect(Collectors.toList());
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+        Thread.sleep(60000);
```

Comment:
Sorry for the confusion , this is actually needed. this is the time it takes for writes to continue after scale. So the writers need to wait atleast until this much time.
Relevant logs to prove the same
```
2017-06-23 06:35:42,953 19399 [grpc-default-executor-0] DEBUG i.p.c.store.host.ZKHostStore - Found owning host: 172.16.19.3:12345 for containerId: 1
 2017-06-23 06:36:31,061 67507 [grpc-default-executor-0] DEBUG i.p.c.store.host.ZKHostStore - Found owning host: 172.16.19.13:12345 for containerId: 1
```
approx 1min time needed for ownership change.